File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
      int n = nums.length, left = 0, right, p, leftSq, rightSq;
      right = p = n - 1;
      int[] result = new int[n];

        while(left <= right) {
          leftSq = nums[left] * nums[left];
          rightSq = nums[right] * nums[right];

          if (leftSq > rightSq) {
            result[p--] = leftSq;
            left++;
          } else {
            result[p--] = rightSq;
            right--;
          }
        }

        return result;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)

        left = 0
        k = right = len(nums) - 1

        while left <= right:
          left_sq = nums[left] * nums[left]
          right_sq = nums[right] * nums[right]

          if (left_sq > right_sq):
            result[k] = left_sq
            left += 1
            k -=1 
          else:
            result[k] = right_sq
            right -= 1
            k -= 1
        
        return result